File: scripts/add-i18n-keys.py
"""
Add missing i18n keys for:
- Stage of Change options (intake)
- Eigen Regie intake options (intake)
- VSP zone labels (prechat-vsp)
- GDPR consent hardcoded strings
- Disclaimer modal (chat)
"""
import json
import os

LOCALES_DIR = '/home/ubuntu/recofree-app/lib/i18n/locales'

# New keys to add
NEW_KEYS = {
    # ─── Stage of Change Options ───────────────────────────────
    "intake.stage.precontemplation.label": {
        "nl": "Nog niet klaar",
        "en": "Not ready yet",
        "fr": "Pas encore prêt"
    },
    "intake.stage.precontemplation.description": {
        "nl": "Ik weet niet zeker of ik nu iets moet veranderen.",
        "en": "I'm not sure I need to change anything right now.",
        "fr": "Je ne suis pas sûr d'avoir besoin de changer quoi que ce soit maintenant."
    },
    "intake.stage.contemplation.label": {
        "nl": "Erover nadenken",
        "en": "Thinking about it",
        "fr": "J'y réfléchis"
    },
    "intake.stage.contemplation.description": {
        "nl": "Ik begin na te denken over verandering.",
        "en": "I'm starting to think about making a change.",
        "fr": "Je commence à penser à faire un changement."
    },
    "intake.stage.preparation.label": {
        "nl": "Ik bereid me voor",
        "en": "Getting ready",
        "fr": "Je me prépare"
    },
    "intake.stage.preparation.description": {
        "nl": "Ik ben van plan om binnenkort actie te ondernemen.",
        "en": "I'm planning to take action soon.",
        "fr": "Je prévois de passer à l'action bientôt."
    },
    "intake.stage.action.label": {
        "nl": "Ik doe het",
        "en": "Taking action",
        "fr": "Je passe à l'action"
    },
    "intake.stage.action.description": {
        "nl": "Ik werk nu actief aan verandering.",
        "en": "I'm actively working on change right now.",
        "fr": "Je travaille activement à changer en ce moment."
    },
    "intake.stage.maintenance.label": {
        "nl": "Ik houd het vol",
        "en": "Maintaining",
        "fr": "Je maintiens"
    },
    "intake.stage.maintenance.description": {
        "nl": "Ik heb veranderingen aangebracht en werk eraan om ze te behouden.",
        "en": "I've made changes and I'm working to keep them.",
        "fr": "J'ai fait des changements et je travaille à les maintenir."
    },

    # ─── Eigen Regie Intake Options ────────────────────────────
    "intake.eigen_regie.1.label": {
        "nl": "Mijn leven draait volledig om de ander",
        "en": "My life revolves entirely around the other person",
        "fr": "Ma vie tourne entièrement autour de l'autre personne"
    },
    "intake.eigen_regie.2.label": {
        "nl": "Ik ben vooral bezig met de ander",
        "en": "I am mostly focused on the other person",
        "fr": "Je suis principalement concentré sur l'autre personne"
    },
    "intake.eigen_regie.3.label": {
        "nl": "Er is een mix tussen mezelf en de ander",
        "en": "There is a mix between myself and the other person",
        "fr": "Il y a un mélange entre moi et l'autre personne"
    },
    "intake.eigen_regie.4.label": {
        "nl": "Ik behoud grotendeels mijn eigen koers",
        "en": "I mostly maintain my own direction",
        "fr": "Je maintiens principalement ma propre direction"
    },
    "intake.eigen_regie.5.label": {
        "nl": "Ik leef volledig mijn eigen leven",
        "en": "I fully live my own life",
        "fr": "Je vis pleinement ma propre vie"
    },

    # ─── VSP Zone Labels (prechat-vsp option titles) ───────────
    "prechat_vsp.option.groen.label": {
        "nl": "Geen spanning",
        "en": "No tension",
        "fr": "Pas de tension"
    },
    "prechat_vsp.option.geel.label": {
        "nl": "Lichte spanning",
        "en": "Mild tension",
        "fr": "Légère tension"
    },
    "prechat_vsp.option.oranje.label": {
        "nl": "Hogere spanning / tijd om in te grijpen",
        "en": "Higher tension / time to intervene",
        "fr": "Tension plus forte / temps d'intervenir"
    },
    "prechat_vsp.option.rood.label": {
        "nl": "Terugval nabij / actie nodig",
        "en": "Relapse near / action needed",
        "fr": "Rechute proche / action nécessaire"
    },
    "prechat_vsp.option.paars.label": {
        "nl": "Terugval",
        "en": "Relapse",
        "fr": "Rechute"
    },

    # ─── GDPR Consent hardcoded strings ────────────────────────
    "gdpr_consent.card.body_intro": {
        "nl": "RecoFree gebruikt AI-technologie (OpenAI) om gesprekken te verwerken.",
        "en": "RecoFree uses AI technology (OpenAI) to process conversations.",
        "fr": "RecoFree utilise la technologie IA (OpenAI) pour traiter les conversations."
    },
    "gdpr_consent.card.body_footer": {
        "nl": "Je persoonlijke gegevens blijven op je toestel. RecoFree slaat geen persoonlijke gegevens op externe servers op.",
        "en": "Your personal data stays on your device. RecoFree does not store personal data on external servers.",
        "fr": "Tes données personnelles restent sur ton appareil. RecoFree ne stocke pas de données personnelles sur des serveurs externes."
    },
    "gdpr_consent.contact_text": {
        "nl": "Vragen? privacy@recofree.app",
        "en": "Questions? privacy@recofree.app",
        "fr": "Questions ? privacy@recofree.app"
    },

    # ─── Chat Disclaimer Modal ─────────────────────────────────
    # chat.modal.title, chat.modal.body, chat.modal.button already exist in all 3 locales
    # But the component uses HARDCODED strings instead of t() — we just need to fix the component

    # ─── Backpack Stage of Change option labels (used in backpack.tsx) ───
    "backpack.stage.precontemplation.label": {
        "nl": "Nog niet klaar",
        "en": "Not ready yet",
        "fr": "Pas encore prêt"
    },
    "backpack.stage.precontemplation.description": {
        "nl": "Ik weet niet zeker of ik nu iets moet veranderen.",
        "en": "I'm not sure I need to change anything right now.",
        "fr": "Je ne suis pas sûr d'avoir besoin de changer quoi que ce soit maintenant."
    },
    "backpack.stage.contemplation.label": {
        "nl": "Erover nadenken",
        "en": "Thinking about it",
        "fr": "J'y réfléchis"
    },
    "backpack.stage.contemplation.description": {
        "nl": "Ik begin na te denken over verandering.",
        "en": "I'm starting to think about making a change.",
        "fr": "Je commence à penser à faire un changement."
    },
    "backpack.stage.preparation.label": {
        "nl": "Ik bereid me voor",
        "en": "Getting ready",
        "fr": "Je me prépare"
    },
    "backpack.stage.preparation.description": {
        "nl": "Ik ben van plan om binnenkort actie te ondernemen.",
        "en": "I'm planning to take action soon.",
        "fr": "Je prévois de passer à l'action bientôt."
    },
    "backpack.stage.action.label": {
        "nl": "Ik doe het",
        "en": "Taking action",
        "fr": "Je passe à l'action"
    },
    "backpack.stage.action.description": {
        "nl": "Ik werk nu actief aan verandering.",
        "en": "I'm actively working on change right now.",
        "fr": "Je travaille activement à changer en ce moment."
    },
    "backpack.stage.maintenance.label": {
        "nl": "Ik houd het vol",
        "en": "Maintaining",
        "fr": "Je maintiens"
    },
    "backpack.stage.maintenance.description": {
        "nl": "Ik heb veranderingen aangebracht en werk eraan om ze te behouden.",
        "en": "I've made changes and I'm working to keep them.",
        "fr": "J'ai fait des changements et je travaille à les maintenir."
    },
}

def add_keys_to_locale(locale_code):
    filepath = os.path.join(LOCALES_DIR, f'{locale_code}.json')
    with open(filepath, 'r') as f:
        data = json.load(f)
    
    added = 0
    for key, translations in NEW_KEYS.items():
        if key not in data:
            data[key] = translations[locale_code]
            added += 1
    
    with open(filepath, 'w') as f:
        json.dump(data, f, ensure_ascii=False, indent=2)
        f.write('\n')
    
    print(f'{locale_code}.json: added {added} new keys (total: {len(data)})')

add_keys_to_locale('nl')
add_keys_to_locale('en')
add_keys_to_locale('fr')
print('\nDone!')

File: scripts/add-i18n-keys.mjs
/**
 * Script to add missing i18n keys to all three locale files.
 * Run with: node scripts/add-i18n-keys.mjs
 */
import { readFileSync, writeFileSync } from 'fs';
import { resolve } from 'path';

const LOCALES_DIR = resolve(import.meta.dirname, '../lib/i18n/locales');

function loadJson(file) {
  return JSON.parse(readFileSync(resolve(LOCALES_DIR, file), 'utf8'));
}

function saveJson(file, data) {
  writeFileSync(resolve(LOCALES_DIR, file), JSON.stringify(data, null, 2) + '\n', 'utf8');
}

// New keys to add
const NEW_KEYS = {
  // Slider title labels (used in mood screen slider cards)
  "mood.slider.craving.label": { en: "Craving", nl: "Verlangen", fr: "Envie" },
  "mood.slider.frustration.label": { en: "Frustration", nl: "Frustratie", fr: "Frustration" },
  "mood.slider.despondency.label": { en: "Despondency", nl: "Moedeloosheid", fr: "Découragement" },
  "mood.slider.focus.label": { en: "Mental Focus", nl: "Mentale focus", fr: "Concentration mentale" },
  "mood.slider.stress.label": { en: "Stress", nl: "Stress", fr: "Stress" },
  "mood.slider.boundaryFatigue.label": { en: "Boundary Fatigue", nl: "Grensmoeheid", fr: "Fatigue des limites" },
  "mood.slider.emotionalBurden.label": { en: "Emotional Burden", nl: "Emotionele last", fr: "Charge émotionnelle" },
  "mood.slider.selfCare.label": { en: "Self-care", nl: "Zelfzorg", fr: "Soin de soi" },

  // Mood alert messages (with {labels} param)
  "mood.alert.severe.message": { en: "{labels} is at a critical level.", nl: "{labels} staat op een kritiek niveau.", fr: "{labels} est à un niveau critique." },
  "mood.alert.severe.message_plural": { en: "{labels} are at a critical level.", nl: "{labels} staan op een kritiek niveau.", fr: "{labels} sont à un niveau critique." },
  "mood.alert.moderate.message": { en: "{labels} is elevated.", nl: "{labels} is verhoogd.", fr: "{labels} est élevé." },
  "mood.alert.moderate.message_plural": { en: "{labels} are elevated.", nl: "{labels} zijn verhoogd.", fr: "{labels} sont élevés." },

  // Mood recognition section
  "mood.recognition.based_on_checkins": { en: "Based on {count} check-in{plural} this week", nl: "Gebaseerd op {count} check-in{plural} deze week", fr: "Basé sur {count} check-in{plural} cette semaine" },

  // Progress card
  "progress_card.signal.softened_prefix": { en: "Softened: ", nl: "Verzacht: ", fr: "Adouci : " },
  "progress_card.signal.increased_prefix": { en: "Needs attention: ", nl: "Vraagt aandacht: ", fr: "Demande attention : " },
  "progress_card.elias.inner_landscape.hope_prefix": { en: "Hope: ", nl: "Hoop: ", fr: "Espoir : " },
  "progress_card.elias.inner_landscape.fear_prefix": { en: "Fear: ", nl: "Angst: ", fr: "Peur : " },

  // Mood trend chart card - check-in recorded
  "mood_trend_chart_card.fallback.count_recorded": { en: "{count} check-in{plural} recorded", nl: "{count} check-in{plural} geregistreerd", fr: "{count} check-in{plural} enregistré{plural2}" },

  // Backpack section labels (Elias)
  "backpack.section.childhood.label": { en: "Childhood", nl: "Kindertijd", fr: "Enfance" },
  "backpack.section.childhood.age_range": { en: "6–12 years", nl: "6–12 jaar", fr: "6–12 ans" },
  "backpack.section.childhood.prompt": { en: "Where did you grow up during this period? Describe the atmosphere at home, your school years, friendships, and events that made an impression on you.", nl: "Waar groeide je op in deze periode? Beschrijf de sfeer thuis, je schooljaren, vriendschappen en gebeurtenissen die indruk op je maakten.", fr: "Où as-tu grandi pendant cette période ? Décris l'atmosphère à la maison, tes années d'école, tes amitiés et les événements qui t'ont marqué." },
  "backpack.section.adolescence.label": { en: "Adolescence", nl: "Adolescentie", fr: "Adolescence" },
  "backpack.section.adolescence.age_range": { en: "12–18 years", nl: "12–18 jaar", fr: "12–18 ans" },
  "backpack.section.adolescence.prompt": { en: "How was your teenage years? How were things at home, at school, and with peers? Did you have struggles or moments of growth?", nl: "Hoe was je tienertijd? Hoe ging het thuis, op school en met leeftijdsgenoten? Had je worstelingen of groeimomenten?", fr: "Comment étaient tes années d'adolescence ? Comment ça se passait à la maison, à l'école et avec tes pairs ? As-tu eu des difficultés ou des moments de croissance ?" },
  "backpack.section.adulthood.label": { en: "Adulthood", nl: "Volwassenheid", fr: "Âge adulte" },
  "backpack.section.adulthood.age_range": { en: "18–50 years", nl: "18–50 jaar", fr: "18–50 ans" },
  "backpack.section.adulthood.prompt": { en: "What are important choices or events in your adult life? Think about work, relationships, children, addiction, loss, growth, or meaning.", nl: "Wat zijn belangrijke keuzes of gebeurtenissen in je volwassen leven? Denk aan werk, relaties, kinderen, verslaving, verlies, groei of betekenis.", fr: "Quels sont les choix ou événements importants de ta vie adulte ? Pense au travail, aux relations, aux enfants, à l'addiction, à la perte, à la croissance ou au sens." },
  "backpack.section.family.label": { en: "Family", nl: "Familie", fr: "Famille" },
  "backpack.section.family.age_range": { en: "Throughout life", nl: "Heel het leven", fr: "Tout au long de la vie" },
  "backpack.section.family.prompt": { en: "How has your relationship with your parents or family been? Are there patterns, loyalties, or tensions that still influence you today?", nl: "Hoe is je relatie met je ouders of familie geweest? Zijn er patronen, loyaliteiten of spanningen die je vandaag nog beïnvloeden?", fr: "Comment a été ta relation avec tes parents ou ta famille ? Y a-t-il des schémas, des loyautés ou des tensions qui t'influencent encore aujourd'hui ?" },
  "backpack.section.themes.label": { en: "Recurring Themes", nl: "Terugkerende thema's", fr: "Thèmes récurrents" },
  "backpack.section.themes.age_range": { en: "Across all phases", nl: "Door alle fases heen", fr: "À travers toutes les phases" },
  "backpack.section.themes.prompt": { en: "Are there recurring themes, beliefs, or inner struggles that you recognize across these life phases?", nl: "Zijn er terugkerende thema's, overtuigingen of innerlijke worstelingen die je herkent door deze levensfasen heen?", fr: "Y a-t-il des thèmes récurrents, des croyances ou des luttes intérieures que tu reconnais à travers ces phases de vie ?" },
  "backpack.section.vsp.label": { en: "Safety Plan (VSP)", nl: "Veiligheidsplan (VSP)", fr: "Plan de sécurité (VSP)" },
  "backpack.section.vsp.age_range": { en: "Personal signals", nl: "Persoonlijke signalen", fr: "Signaux personnels" },
  "backpack.section.vsp.prompt": { en: "Write your safety plan here per zone.", nl: "Schrijf je veiligheidsplan hier per zone.", fr: "Écris ton plan de sécurité ici par zone." },

  // Backpack progress counter
  "backpack.progress.counter": { en: "{filled} of {total} sections", nl: "{filled} van {total} onderdelen", fr: "{filled} sur {total} sections" },

  // Kim backpack section labels
  "backpack.kim.my_story.title": { en: "My Story", nl: "Mijn verhaal", fr: "Mon histoire" },
  "backpack.kim.my_story.subtitle": { en: "Who am I outside of this relationship?", nl: "Wie ben ik buiten deze relatie?", fr: "Qui suis-je en dehors de cette relation ?" },
  "backpack.kim.the_relationship.title": { en: "The Relationship", nl: "De relatie", fr: "La relation" },
  "backpack.kim.the_relationship.subtitle": { en: "How did it evolve? When did it change?", nl: "Hoe is het geëvolueerd? Wanneer veranderde het?", fr: "Comment a-t-elle évolué ? Quand a-t-elle changé ?" },
  "backpack.kim.the_impact.title": { en: "The Impact", nl: "De impact", fr: "L'impact" },
  "backpack.kim.the_impact.subtitle": { en: "What has addiction done to my life, family, work?", nl: "Wat heeft verslaving gedaan met mijn leven, familie, werk?", fr: "Qu'est-ce que l'addiction a fait à ma vie, ma famille, mon travail ?" },
  "backpack.kim.my_boundaries.title": { en: "My Boundaries", nl: "Mijn grenzen", fr: "Mes limites" },
  "backpack.kim.my_boundaries.subtitle": { en: "What can I carry? What have I already tried?", nl: "Wat kan ik dragen? Wat heb ik al geprobeerd?", fr: "Que puis-je porter ? Qu'ai-je déjà essayé ?" },
  "backpack.kim.my_strength.title": { en: "My Strength", nl: "Mijn kracht", fr: "Ma force" },
  "backpack.kim.my_strength.subtitle": { en: "Where do I find strength? What do I want for myself?", nl: "Waar vind ik kracht? Wat wil ik voor mezelf?", fr: "Où est-ce que je trouve ma force ? Que veux-je pour moi-même ?" },

  // Profile STAGE_LABELS
  "profile.stage.precontemplation": { en: "Precontemplation", nl: "Voorbeschouwing", fr: "Précontemplation" },
  "profile.stage.contemplation": { en: "Contemplation", nl: "Beschouwing", fr: "Contemplation" },
  "profile.stage.preparation": { en: "Preparation", nl: "Voorbereiding", fr: "Préparation" },
  "profile.stage.action": { en: "Action", nl: "Actie", fr: "Action" },
  "profile.stage.maintenance": { en: "Maintenance", nl: "Onderhoud", fr: "Maintien" },

  // Profile stats
  "profile.user_card.stats.line": { en: "{companion} · {stage} · {sessions} session{sessionPlural} · {checkins} check-in{checkinPlural}", nl: "{companion} · {stage} · {sessions} sessie{sessionPlural} · {checkins} check-in{checkinPlural}", fr: "{companion} · {stage} · {sessions} session{sessionPlural} · {checkins} check-in{checkinPlural}" },
  "profile.user_card.stats.line_kim": { en: "{companion} · {sessions} session{sessionPlural} · {checkins} check-in{checkinPlural}", nl: "{companion} · {sessions} sessie{sessionPlural} · {checkins} check-in{checkinPlural}", fr: "{companion} · {sessions} session{sessionPlural} · {checkins} check-in{checkinPlural}" },
  "profile.user_card.stats.plural_s": { en: "s", nl: "s", fr: "s" },

  // Profile emergency contact remove
  "profile.emergency_contacts.alert.remove.message_name": { en: "Remove {name}?", nl: "{name} verwijderen?", fr: "Supprimer {name} ?" },

  // VSP export dialog title
  "profile.vsp_insight.share_title": { en: "VSP Insight Overview", nl: "VSP Insight Overzicht", fr: "Aperçu VSP Insight" },

  // Eigen Regie labels
  "mood.eigen_regie.question": { en: "To what extent was your day today determined by the choices of the other person?", nl: "In hoeverre werd je dag vandaag bepaald door de keuzes van de ander?", fr: "Dans quelle mesure ta journée a-t-elle été déterminée par les choix de l'autre personne ?" },
  "mood.eigen_regie.slider.min": { en: "Full self-direction", nl: "Volledige eigen regie", fr: "Pleine autonomie" },
  "mood.eigen_regie.slider.max": { en: "Fully determined by the other", nl: "Volledig bepaald door de ander", fr: "Entièrement déterminé par l'autre" },
  "mood.eigen_regie.zone.ROOD": { en: "Red", nl: "Rood", fr: "Rouge" },
  "mood.eigen_regie.zone.ORANJE": { en: "Orange", nl: "Oranje", fr: "Orange" },
  "mood.eigen_regie.zone.GEEL": { en: "Yellow", nl: "Geel", fr: "Jaune" },
  "mood.eigen_regie.zone.LICHTGROEN": { en: "Light Green", nl: "Lichtgroen", fr: "Vert clair" },
  "mood.eigen_regie.zone.GROEN": { en: "Green", nl: "Groen", fr: "Vert" },
  "mood.eigen_regie.meaning.ROOD": { en: "I was completely focused on the other person. I felt responsible for their behavior.", nl: "Ik was volledig gericht op de ander. Ik voelde me verantwoordelijk voor hun gedrag.", fr: "J'étais entièrement concentré sur l'autre personne. Je me sentais responsable de son comportement." },
  "mood.eigen_regie.meaning.ORANJE": { en: "I was mostly occupied with the other person. My own needs barely came up.", nl: "Ik was vooral bezig met de ander. Mijn eigen behoeften kwamen nauwelijks aan bod.", fr: "J'étais principalement occupé par l'autre personne. Mes propres besoins n'ont presque pas été abordés." },
  "mood.eigen_regie.meaning.GEEL": { en: "I was often thinking about the other person, but also thought about myself briefly.", nl: "Ik dacht vaak aan de ander, maar dacht ook kort aan mezelf.", fr: "Je pensais souvent à l'autre personne, mais j'ai aussi brièvement pensé à moi-même." },
  "mood.eigen_regie.meaning.LICHTGROEN": { en: "I considered the other person, but also stayed with myself.", nl: "Ik hield rekening met de ander, maar bleef ook bij mezelf.", fr: "J'ai tenu compte de l'autre personne, mais je suis aussi resté avec moi-même." },
  "mood.eigen_regie.meaning.GROEN": { en: "I followed my own plan. I felt free, regardless of what the other person did.", nl: "Ik volgde mijn eigen plan. Ik voelde me vrij, ongeacht wat de ander deed.", fr: "J'ai suivi mon propre plan. Je me sentais libre, peu importe ce que l'autre personne faisait." },
};

// Load all three locale files
const en = loadJson('en.json');
const nl = loadJson('nl.json');
const fr = loadJson('fr.json');

// Add new keys
let addedCount = 0;
for (const [key, translations] of Object.entries(NEW_KEYS)) {
  if (!en[key]) { en[key] = translations.en; addedCount++; }
  if (!nl[key]) { nl[key] = translations.nl; }
  if (!fr[key]) { fr[key] = translations.fr; }
}

// Save
saveJson('en.json', en);
saveJson('nl.json', nl);
saveJson('fr.json', fr);

console.log(`Done. Added ${addedCount} new keys to locale files.`);
